src/Components/DrawingCanvas.jsx

```diff
@@ -0,0 +1,308 @@
+          drawingHistory.push(drawCanvas.toDataURL());
+        }
+      };
+    
+      const drawRectangle = (x, y) => {
+        const { context, drawCanvas } = getContext();
+    
+        const width = x - rectStart.x;
+        const height = y - rectStart.y;
+      
+        context.clearRect(0, 0, drawCanvas.width, drawCanvas.height);
+        context.strokeRect(rectStart.x, rectStart.y, width, height);
+
+        if(document.getElementById('solid-fill').checked) {
+          context.fillRect(rectStart.x, rectStart.y, width, height);
+        };
+      }
+    
+      const drawEllipse = (x, y) => {
+        const { context, drawCanvas } = getContext();
+    
+        const width = x - rectStart.x;
+        const height = y - rectStart.y;
+    
+        const centerX = rectStart.x + width / 2;
+        const centerY = rectStart.y + height / 2;
+        const radiusX = Math.abs(width / 2);
+        const radiusY = Math.abs(height / 2);
+    
+        context.clearRect(0, 0, drawCanvas.width, drawCanvas.height);
+        context.beginPath();
+        context.ellipse(centerX, centerY, radiusX, radiusY, 0, 0, 2 * Math.PI);
+        context.stroke();
+
+        if(document.getElementById('solid-fill').checked) {
+          context.fill();
+        };
+      }
+    
+      const drawTriangle = (x, y) => {
+        const { context , drawCanvas } = getContext();
+    
+        const sideLength = Math.sqrt(Math.pow(x - rectStart.x, 2) + Math.pow(y - rectStart.y, 2));
+    
+        const angle = Math.atan2(y - rectStart.y, x - rectStart.x);
+    
+        const angleOffset = Math.PI / 9;
+    
+        const x2 = rectStart.x + sideLength * Math.cos(angle + angleOffset);
+        const y2 = rectStart.y + sideLength * Math.sin(angle + angleOffset);
+    
+        const x3 = rectStart.x + sideLength * Math.cos(angle - angleOffset);
+        const y3 = rectStart.y + sideLength * Math.sin(angle - angleOffset);
+    
+        context.clearRect(0, 0, drawCanvas.width, drawCanvas.height);
+        context.beginPath();
+        context.moveTo(rectStart.x, rectStart.y);
+        context.lineTo(x2, y2);
+        context.lineTo(x3, y3);
+        context.closePath();
+        context.stroke();
+
+        if(document.getElementById('solid-fill').checked) {
+          context.fill();
+        };
+      }
+    
+      const draw = (isTouch) => (e) => {
+        if (!drawing) return;
+    
+        if(document.getElementById('pen').checked) {
+          colorPicker();
+        }
+    
+        const { drawCanvas, context } = getContext();
+    
+        let position;
+    
+        if (isTouch) {
+          position = getTouchPos(drawCanvas, e);
+        } else {
+          position = getMousePos(drawCanvas, e);
+        }
+          
+        let { x , y } = position;
+    
+        if (square.checked) {
+          drawRectangle(x, y);
+        
+        } else if (ellipse.checked) {
+          drawEllipse(x,y);
+    
+        } else if (triangle.checked) {
```

Comment:
`square`, `ellipse` și `triangle` nu sunt definite aici. Poate ai copiat codul de altundeva?

Reply:
Sunt id-urile de la forme, au ramas asa de cand aveam toate in acelasi fisier si nu am vazut sa imi arunce erori, asa ca am lasat asa. Am modificat in "getElementbyId".